### DISCUSSION TITLE
Neptune integration

### DISCUSSION THREAD

**Discussion by MatejNedic** [awspring/spring-cloud-aws#36] [Ideas]
Neptune is graph based database. Currently there is no Spring Cloud integration for Neptune.

https://aws.amazon.com/neptune/
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/neptune/AmazonNeptuneClient.html
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/neptune/AmazonNeptune.html#describeDBInstances-com.amazonaws.services.neptune.model.DescribeDBInstancesRequest-

**Comment by eddumelendez** [comment]
What would be the scope of this? Just to set up the client? I think that if there is a `spring-data-neptune` project there would be something to do in the project. This is similar to `spring-data-cosmodb` [here](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/cosmos/azure-spring-data-cosmos)

Also, as far as any integration helps from the application side would be great to add it.

**Reply by MatejNedic** [reply]
@eddumelendez was thinking about full integration as Spring-Data-cosmodb has or `spring-data-dynamodb` [here](https://github.com/derjust/spring-data-dynamodb). I know it is a lot of work but it would make sense once when all things are moved to SDK 2 and some integrations are rewritten to look for new integrations. This might not be the most popular integration though, but just an idea to have, that this integration can be made if there is interest in it. What is your opinion? Is it worth the trouble?

**Reply by eddumelendez** [reply]
make sense to do that but also let's see how often this service is used to make it a priority.
